MACOSX_NOTES

```diff
@@ -19,6 +19,37 @@
 
 brew install git mysql exiftool libjpeg shared-mime-info openssl imagemagick findutils
 
+# If you haven't done so already, install a more recent version of Bash and set
+# it as the default. You can find your installed version like this:
+# bash --version
+
+# Apple only includes Bash 3.2 from 2007 in all versions of Mac OS X even now,
+# because Bash > 4.0 uses GPLv3, and they don't want to support that license.
+# MO's scripts use syntax that requires Bash >= 4.0.
+# Reference: https://medium.com/p/7138bd1066ba
+# https://stackoverflow.com/questions/6047648/associative-arrays-error-declare-a-invalid-option
+# To install the newest Bash alongside older versions:
+# brew install bash
+
+# now check paths of all versions installed
+# which -a bash
+# you should get something like this
+# /usr/local/bin/bash
+# /bin/bash
+# check those versions
+# /usr/local/bin/bash --version
+# /bin/bash --version
+# now check
+# bash --version
+# should be the new one.
+# So far, so good. Now you have make this version the default.
+# sudo vim /etc/shells
+# (add at the bottom of the list: /usr/local/bin/bash and save out)
+# Set the default shell:
+# chsh -s /usr/local/bin/bash
+# Set the default shell for the root user:
+# sudo chsh -s /usr/local/bin/bash
```

Comment:
This is not needed.  I use zsh (the default in current MacOSX) and don't have any issues.

Reply:
Huh. OK, I use zsh too. 

I had believed this was the line that made the script finally work, but i may have been "over-hammering the nail" at that point and not paying close enough attention. In any case -- does this line change the shell app? My impression was it just told it which environment to run in.

Reply:
Turned it into a note reference.

Reply:
@mo-nathan Please feel free to edit further.

Reply:
I think `chsh` changes your default shell.  So after running that I don't think you'll be using zsh any more.  Probably the better solution would be to tweak the scripts so they don't care which version of bash they are using.  Looks like you could use $BASH_VERSION to make this decision.  While the whole Apple bash/zsh thing is dumb, it's even dumber than the new version of bash is not backwards compatible with the old version.  I appreciate they are trying to make a point, but breaking the world is not a good way to make new friends.

Reply:
Actually looking into it that may not be accurate.  I'm not sure old bash had support for associative arrays.  So maybe it is backwards compatible.  Maybe the scripts should just be converted to ruby or python.

Reply:
That's right. That's the whole reason for the glitch - Apple's ancient Bash wouldn't make an associative array.

Reply:
Clarification - of course you're right, i'm in fact running the new bash, not zsh, after running that command. I didn't notice because the prompt style of the updated bash is just like zsh (but not like the old bash), and i didn't check the window frame, which clearly says i'm running bash!